File: Arrays_101/Remove Element/solution.cpp
// class Solution {
// public:
//     int removeElement(vector<int>& nums, int val)
//     {
//         vector<int>::iterator it=nums.begin();
//         while (it !=nums.end())
//         {
//             if (*it==val)
//                 it = nums.erase(it);
//             else
//                 it++;
//         }
//         return nums.size();
//     }
// };

// OR

class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        vector<int> res;
        int k=0;

        for(int i=0; i<nums.size(); i++){
            
            if(nums[i]==val){
                continue;
            }

            res.push_back(nums[i]);
            k++;
    
        }
      
        for(int i=0;i<res.size();i++)
            nums[i]= res[i];

        return k;
    }
};

File: Arrays_101/Remove Element/solution.js
// /**
//  * @param {number[]} nums
//  * @param {number} val
//  * @return {number}
//  */
//  var removeElement = function(nums, val) {
//     let index = 0
//     for (let i = 0; i < nums.length; i++) {
//         if (nums[i] !== val) {
//             nums[index] = nums[i]
//             index++
//         }
//     }
//     return index
// };

// OR

/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function (nums, val) {
  var res = nums.filter((num) => num !== val);
  for (let i = 0; i < res.length; i++) {
    nums[i] = res[i];
  }
  return res.length;
};
